### PR TITLE
Support for non-public clouds

### PR DESCRIPTION
**Pull request by seguler**
Issue #22 - Adds support for non-public clouds via the use of account URI from connection.cfg

Here is connection.cfg template:
```
accountName myaccount
accountKey mykey==
containerName mycontainer
blobEndpoint myaccount.blob.core.windows.net
```

### CODE REVIEW

**Comment by seguler** [comment]
changed accountUri to blobEndpoint @asorrin

File: azure-storage-cpp-lite/src/storage_account.cpp [hunk 1]
```diff
@@ -5,7 +5,8 @@
 namespace microsoft_azure {
     namespace storage {
 
-        storage_account::storage_account(const std::string &account_name, std::shared_ptr<storage_credential> credential, bool use_https, const std::string &domain_suffix)
+        // TODO: Clean up table queue and file services
+        storage_account::storage_account(const std::string &account_name, std::shared_ptr<storage_credential> credential, bool use_https, const std::string &blob_endpoint)
```
Comment (AlexanderYukhanov) [inline azure-storage-cpp-lite/src/storage_account.cpp]: oh, what a strange way to populate and handle those variables

Comment (seguler) [reply azure-storage-cpp-lite/src/storage_account.cpp]: Are you saying 'passing by reference' is odd ? We can address this later... I am only adding blob_endpoint here..

Comment (AlexanderYukhanov) [reply azure-storage-cpp-lite/src/storage_account.cpp]: no -) i am saying that this class is doing a very simple thing in an unexpectedly complex way -) I ok with your changes.

**Pull request merged**